### PR TITLE
fix(vite,webpack): disable async entrypoint by default

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Previously, when running vite, the following error would be printed in dev mode:

```
Uncaught (in promise) TypeError: error loading dynamically imported module
```

This was a side effect of https://github.com/nuxt/framework/pull/2223.

This PR disables async entry by default, though it can be re-enabled with a flag. **Note**: you may well find that you need to disable async entry in development.

This PR also swaps entry <-> bootstrap names (as I think it makes more sense this way). Despite the massive git diff, the only real change is to the webpack/vite configs.

### 👉 Migration

If you were previously relying on the async entrypoint behaviour (for example, with module federation) then you can reenable previous behaviour by adding `experimental.asyncEntry` to your nuxt.config:

```diff
  import { defineNuxtConfig } from 'nuxt3'
  
  export default defineNuxtConfig({
+   experimental: {
+     asyncEntry: true // or process.env.NODE_ENV !== 'development'
+   }
  })
```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
I believe we could revert async entry change back to normal for all targets (dev/prod x vite/webpack) and optionally support async method behind a flag.

File: packages/schema/src/config/build.ts [hunk 1]
```diff
@@ -617,5 +617,11 @@ export default {
...
    * @version 2
    */
-  followSymlinks: false
+
+  followSymlinks: false,
+  /**
+   * Set to true to generate an async entry (for module federation support).
+   * @version 3
+   */
+  asyncEntry: false
```
[the remark this answers is not in the document]
Comment (danielroe) [reply packages/schema/src/config/build.ts]: I like experimental namespace 👍

File: packages/vite/src/vite.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export async function bundle (nuxt: Nuxt) {
             // will be filled in client/server configs
             '#build/plugins': '',
             '#build': nuxt.options.buildDir,
-            '/entry.mjs': resolve(nuxt.options.appDir, 'entry'),
+            '/entry.mjs': resolve(nuxt.options.appDir, nuxt.options.dev || !nuxt.options.experimental.asyncEntry ? 'entry' : 'entry.async'),
```
Comment (pi0) [inline packages/vite/src/vite.ts:53]: Any reason to add . dev check still? Considering async entry is experimental we could let people trying async entry issues

File: packages/webpack/src/presets/base.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ function baseConfig (ctx: WebpackConfigContext) {
 
   ctx.config = {
     name: ctx.name,
-    entry: { app: [resolve(options.appDir, 'entry')] },
+    entry: { app: [options.experimental.asyncEntry ? resolve(options.appDir, 'entry.async') : resolve(options.appDir, 'entry')] },
```
Comment (pi0) [inline packages/webpack/src/presets/base.ts:26]: Refactor: we could shorten logic by adding asyncEntry chrck inside template literal:

```suggestion
    entry: { app: [resolve(options.appDir,  options.experimental.asyncEntry? 'entry.async' : 'entry') ] },
```

File: packages/vite/src/vite.ts (hunk 2 above)
Comment (danielroe) [reply packages/vite/src/vite.ts:53]:
```suggestion
            '/entry.mjs': resolve(nuxt.options.appDir, nuxt.options.experimental.asyncEntry ? 'entry.async' : 'entry'),
```

**Review by pi0 — APPROVED** [review]

File: packages/vite/src/vite.ts (hunk 2 above)
Comment (danielroe) [reply packages/vite/src/vite.ts]: The only point to make is that we know this will be buggy.

**Comment by pi0** [comment]
(i guess broke linter 🙈)

**Pull request merged**